Source/SwiftLintBuiltInRules/Rules/Lint/AsyncWithoutAwaitRuleExamples.swift

```diff
@@ -0,0 +1,328 @@
+        func bar() async -> Int
+        """),
+        Example("""
+        var foo: Int { get async }
+        var bar: Int { get async }
+        """),
+        Example("""
+        init(foo: bar) async
+        init(baz: qux) async
+        let baz = { qux() }
+        """),
+        Example("""
+        typealias Foo = () async -> Void
+        typealias Bar = () async -> Void
+        let baz = { qux() }
+        """),
+        Example("""
+        func test() async {
+            for await foo in bar {}
+        }
+        """),
+        Example("""
+        func test() async {
+            while let foo = await bar() {}
+        }
+        """),
+        Example("""
+        func test() async {
+            async let foo = bar()
+            let baz = await foo
+        }
+        """),
+        Example("""
+        func test() async {
+            async let foo = bar()
+            await foo
+        }
+        """),
+    ]
+
+    static let triggeringExamples = [
+        Example("""
+        func test() ↓async {
+            perform()
+        }
+        """),
+        Example("""
+        func test() {
+            func baz() ↓async {
+                qux()
+            }
+            perform()
+            func baz() {
+                qux()
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            func baz() async {
+                await qux()
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+          func foo() ↓async {}
+          let bar = { await foo() }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            let bar = {
+                func foo() ↓async {}
+            }
+        }
+        """),
+        Example("let x: (() ↓async -> Void)? = { test() }"),
+        Example("""
+        var test: Int {
+            get ↓async throws {
+                foo()
+            }
+        }
+        """),
+        Example("""
+        var test: Int {
+            get ↓async throws {
+                func foo() ↓async {}
+                let bar = { await foo() }
+            }
+        }
+        """),
+        Example("""
+        var test: Int {
+            get throws {
+                func foo() {}
+                let bar: () ↓async -> Void = { foo() }
+            }
+        }
+        """),
+        Example("init() ↓async {}"),
+        Example("""
+        init() ↓async {
+            func foo() ↓async {}
+            let bar: () ↓async -> Void = { foo() }
+        }
+        """),
+        Example("""
+        subscript(row: Int) -> Double {
+            get ↓async {
+                1.0
+            }
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            for foo in bar {}
+        }
+        """),
+        Example("""
+        func test() ↓async {
+            while let foo = bar() {}
+        }
+        """),
+        Example("""
+        func test() ↓async {
```

Comment:
This example should actually not trigger the rule. `async let x = ...` is only allowed in an `async` context.

Reply:
You are right. I will fix it.